host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -203,49 +216,115 @@ defmodule HostCore.Actors.ActorSupervisor do
     end
   end
 
+  def start_actor_from_file(host_id, fileref, count \\ 1, annotations \\ %{}) do
+    {:ok, {pid, _prefix}} = VirtualHost.lookup(host_id)
+    config = VirtualHost.config(pid)
+
+    if not config.enable_actorfromfs do
+      raise "actor from local filesystem is disabled"
+    end
+
+    Tracer.with_span "Starting Actor from file", kind: :server do
+      case File.read(String.trim_leading(fileref, "file://")) do
+        {:error, err} ->
+          Tracer.add_event("file read failed", reason: "#{inspect(err)}")
+
+          Logger.error(
+            "Failed to read actor file from ${fileref}: #{inspect(err)}",
+            fileref: fileref
+          )
+
+          {:error, err}
+
+        {:ok, binary} ->
+          binary
+          |> start_actor(host_id, fileref, count, annotations)
+      end
+    end
+  end
+
   def live_update(host_id, ref, span_ctx \\ nil) do
-    creds = VirtualHost.get_creds(host_id, :oci, ref)
     {:ok, {pid, lattice_prefix}} = VirtualHost.lookup(host_id)
     config = VirtualHost.config(pid)
 
-    with {:ok, bytes} <-
-           Native.get_oci_bytes(
-             creds,
-             ref,
-             config.allow_latest,
-             config.allowed_insecure
-           ),
-         {:ok, new_claims} <-
-           bytes |> IO.iodata_to_binary() |> Native.extract_claims(),
-         {:ok, old_claims} <-
-           HostCore.Claims.Manager.lookup_claims(lattice_prefix, new_claims.public_key),
-         :ok <- validate_actor_for_update(old_claims, new_claims) do
-      HostCore.Claims.Manager.put_claims(host_id, lattice_prefix, new_claims)
-      HostCore.Refmaps.Manager.put_refmap(host_id, lattice_prefix, ref, new_claims.public_key)
-      targets = find_actor(new_claims.public_key, host_id)
-
-      Logger.info("Performing live update on #{length(targets)} instances",
-        actor_id: new_claims.public_key,
-        oci_ref: ref
-      )
-
-      # Each spawned function is a new process, therefore a new root trace
-      # this is why we pass the span context so all child updates roll up
-      # to the current trace
-      Enum.each(targets, fn pid ->
-        ActorModule.live_update(
-          config,
-          pid,
-          IO.iodata_to_binary(bytes),
-          new_claims,
-          ref,
-          span_ctx
-        )
-      end)
-
-      :ok
-    else
-      err -> {:error, err}
+    cond do
+      String.starts_with?(ref, "file://") ->
+        if not config.enable_actorfromfs do
+          raise "actor from local filesystem is disabled"
+        end
+
+        with {:ok, binary} <- File.read(String.trim_leading(ref, "file://")),
+             {:ok, new_claims} <- binary |> Native.extract_claims(),
```

Comment:
nit: Let's do `Native.extract_claims(binary)` here since the other statements follow that convention rather than the pipe